packages/core/test/unit/sql/literal.test.ts

```diff
@@ -171,3 +171,83 @@ describe('fn', () => {
     });
   });
 });
+
+describe('sql.join', () => {
```

Comment:
It might be worth to include a test with some form of `BaseSqlExpression` here, don't you think?

Reply:
Woops, I was confused by the string array that is columns. You're right on that one.
I was thinking of  the following scenarios;
- parts being a string array
- parts being a `BaseSqlExpression` array (like in the current test)
- separator being a string (like in the current test)
- separator being a `BaseSqlExpession`

Reply:
Good that you mention it, because it raises a design inconsistency

`sql` escapes all string values, but `sql.join` treats them as literals

I think only `sql.literal` should treat strings as literals, and `sql.join` should use the safer approach that `sql` uses